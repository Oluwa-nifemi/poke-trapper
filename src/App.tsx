import React from 'react';
import './App.css';
import {Route, Routes} from "react-router-dom";
import Layout from "./components/layout";
import ListPokemon from "./pages/list-pokemon";
import PokemonDetails from "./pages/pokemon-details";
import MyPokemonList from "./pages/my-pokemon-list";

function App() {
    return (
        <Routes>
            <Route path="/" element={<Layout/>}>
                <Route index element={<ListPokemon/>}/>
                <Route path="/:id" element={<PokemonDetails/>}/>
                <Route path="/my-pokemon" element={<MyPokemonList/>}/>
            </Route>
        </Routes>
    );
}

export default App;
