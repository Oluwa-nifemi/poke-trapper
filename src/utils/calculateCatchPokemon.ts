const calculateCatchPokemon = (): boolean => Math.random() > 0.5

export default calculateCatchPokemon