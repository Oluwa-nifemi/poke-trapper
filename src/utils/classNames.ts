/* This function allows you to merge several classes together.
 * For conditional classes the first item in the array should be the condition (boolean)
 */
export const classNames = (...args: (string | [boolean, string])[]) => {
    return args.filter(className => {
        if (Array.isArray(className)) {
            return className[0];
        }

        return className;
    }).map((className) => {
        if (Array.isArray(className)) {
            return className[1];
        }

        return className;
    }).join(' ')
};