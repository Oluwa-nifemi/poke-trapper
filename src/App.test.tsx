import localforage from "localforage";
import {render, screen} from "tests/utils";
import {waitFor} from "@testing-library/react";
import App from "./App";

describe("Main App Tests", () => {
    it("Renders correct stats from localforage", async () => {
        //Update local storage with data
        await localforage.setItem("caughtPokemon", [
            {
                id: "123",
                nickname: "Test Pokemon",
                sprites: {
                    front_default: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png"
                },
                name: "bulbasaur"
            }
        ])
        await localforage.setItem("stats", {
            bulbasaur: 1
        })

        const {container} = render(<App/>);
        await waitFor(() => expect(screen.queryByText("Loading...")).toBeNull());
        expect(container).toMatchSnapshot();
    });
});