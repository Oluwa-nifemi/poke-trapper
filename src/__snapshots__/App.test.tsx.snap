// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main App Tests Can load main app and view details and add to personal pokemon 1`] = `
<div>
  <nav
    class="nav"
  >
    <a
      href="/"
    >
      <svg
        class="navLogo"
      >
        logo.svg
      </svg>
    </a>
    <a
      class="navLink"
      href="/my-pokemon"
    >
      My Pokemon
    </a>
  </nav>
  <main
    class="main"
  >
    <h1
      class="header"
    >
      List of pokemon
    </h1>
    <section
      class="pokemonRows"
    >
      <article
        class="pokemonRowsHeader"
      >
        <span>
          Name
        </span>
        <span>
          Owned
        </span>
      </article>
      <article
        class="pokemon"
      >
        <a
          class="pokemonName"
          href="/bulbasaur"
        >
          bulbasaur
        </a>
        <span>
          0
        </span>
      </article>
      <article
        class="pokemon"
      >
        <a
          class="pokemonName"
          href="/ivysaur"
        >
          ivysaur
        </a>
        <span>
          0
        </span>
      </article>
      <article
        class="pokemon"
      >
        <a
          class="pokemonName"
          href="/venusaur"
        >
          venusaur
        </a>
        <span>
          0
        </span>
      </article>
    </section>
  </main>
</div>
`;

exports[`Main App Tests Renders correct stats from localforage 1`] = `
<div>
  <nav
    class="nav"
  >
    <a
      href="/"
    >
      <svg
        class="navLogo"
      >
        logo.svg
      </svg>
    </a>
    <a
      class="navLink"
      href="/my-pokemon"
    >
      My Pokemon
    </a>
  </nav>
  <main
    class="main"
  >
    <h1
      class="header"
    >
      List of pokemon
    </h1>
    <section
      class="pokemonRows"
    >
      <article
        class="pokemonRowsHeader"
      >
        <span>
          Name
        </span>
        <span>
          Owned
        </span>
      </article>
      <article
        class="pokemon"
      >
        <a
          class="pokemonName"
          href="/bulbasaur"
        >
          bulbasaur
        </a>
        <span>
          1
        </span>
      </article>
      <article
        class="pokemon"
      >
        <a
          class="pokemonName"
          href="/ivysaur"
        >
          ivysaur
        </a>
        <span>
          0
        </span>
      </article>
      <article
        class="pokemon"
      >
        <a
          class="pokemonName"
          href="/venusaur"
        >
          venusaur
        </a>
        <span>
          0
        </span>
      </article>
    </section>
  </main>
</div>
`;
